Adds a Config for Auto-Deadminning on Ready Up and Latejoining

Jackraxxus opened the pull request:
## About The Pull Request
Fixes #89458
Adds a config for auto-deadminning admins who ready up before the round starts, or click join game after.
The deadminning happens as soon as the button is pressed, not when a job is selected or the round starts. I figure it's better to do it straight away since both have opportunities for metainfo to be posted in admin chat. Before there's admins messing with dynamic config / plotting events etc. and during the round there's basically everything that gets posted to admin logs.

I've had to add a typecheck to the auto_deadmin proc, since the lobby menu technically happens on the centcom Z-level, and there's a pref for ignoring deadminning on centcom Z, so it checks to see if the admin is a new player mob.

Changed the old config from auto_deadmin_player to auto_deadmin_always to make it less deceptive.
Was suggested a pref could be added to do this. I can do that if people want it maybe, but it's not in this PR at the time of posting it.

Is now a pref
<img src="https://i.ibb.co/211sBMYd/Deadmin-Prefs1.png">
<img src="https://i.ibb.co/r20Srbw4/Deadmin-Prefs2.png">

I dunno if the latejoin proccall is in the right spot in the click sequence thing I can move it if people want.
## Why It's Good For The Game

Admins spawning themselves on station is very important for both shenanigans and troubleshooting. It's annoying to have to click readmin every time and I'm too lazy to figure out how to perms escalate my way into disabling that config every round (Though I have tried).

This PR is marginally more likely to convince Timber to turn that off than cussing him out in adminbus.
## Changelog
:cl:
admin: Added a new config to force admins to deadmin when readying up or latejoining the round.
/:cl:

Jackraxxus committed:
Base changes

Contains the basic functionality, config options

Jackraxxus committed:
Legacy Config Options

auto_deadmin_players -> auto_deadmin_always

And I forgot to re-comment the option I changed

Jackraxxus commented:
Works on my testing.
Do NOT ask how many attempts it took before I realised that being a localhost or centcom Z-level prevents auto deadmin from running.

Jackraxxus committed:
Better comment

I had initially considered making the latejoin and ready up things separate procs and forgot to update the comment. Figured it didn't make sense to have them separate.

LemonInTheDark commented on config/config.txt:
```diff
@@ -368,8 +368,11 @@ AUTOADMIN_RANK Game Master
 
-## Uncomment to automatically deadmin players when the game starts.
-#AUTO_DEADMIN_PLAYERS
+## Uncomment to automatically deadmin players whenever they are controlling a mob.
+#AUTO_DEADMIN_ALWAYS
```
@tgstation/head-admins is this something that you think would ever see use over the new config option, here or downstream?

Timberpoes replied:
Who knows re downstreams. 

There's an admin pref for always deadmin, which is the only reason I suggested we keep it in the linked issue report; to retain parity with the pref plus config override paradigm we have going.

I have no particular attachment to an always deadmin config, I only enabled it because I thought it did what this PR adds rather than generating admin tears at the client freeze when readminning.

Jackraxxus committed:
Makes it a pref too

Tho if it's enabled on a config level a pref doesnt really matter. Maybe good for admins on downstreams perhaps.

Jackraxxus commented:
Made it a pref for the whole parity thing. Will update the OP with screenshots.

Timberpoes reviewed (approved):
Uh the code probably works?????? let's try it!
